front/src/applications/editor/Editor.tsx

```diff
@@ -280,13 +281,16 @@ const Editor = () => {
   useEffect(() => {
     if (toolAndState.tool.onMount) toolAndState.tool.onMount(extendedContext);
 
-    const layersList = toolAndState.tool.requiredLayers
-      ? new Set([...editorState.editorLayers, ...toolAndState.tool.requiredLayers])
-      : editorState.editorLayers;
+    const { requiredLayers, incompatibleLayers } = toolAndState.tool;
+    if (requiredLayers || incompatibleLayers) {
```

Comment:
```suggestion
```
I think that you can remove this condition

Reply:
Nice comment, but actually this test avoids dispatching a new action in the store if not needed, so I would prefer to keep it like this if it's ok for you :)